docs/architecture/adr-081-protobuf-mgmt.md

```diff
@@ -0,0 +1,202 @@
+# ADR 081: Protocol Buffers Management
+
+## Changelog
+
+- 2022-02-28: First draft
+
+## Status
+
+Accepted
+
+## Context
+
+At present, we manage the [Protocol Buffers] files ("protos") that define our
+wire-level data formats within the Tendermint repository itself (see the
+[`proto`](../../proto/) directory). Recently, we have been making use of [Buf],
+both locally and in CI, in order to generate Go stubs, and lint and check
+`.proto` files for breaking changes.
+
+The version of Buf used at the time of this decision was `v1beta1`, and it was
+discussed in [\#7975] and in weekly calls as to whether we should upgrade to
+`v1` and harmonize our approach with that used by the Cosmos SDK. The team
+managing the Cosmos SDK was primarily interested in having our protos versioned
+and easily accessible from the [Buf] registry.
+
+The three main customers for the `.proto` files and their needs, as currently
+understood, are as follows.
+
+1. Tendermint needs Go code generated from `.proto` files.
+2. Consumers of Tendermint's `.proto` files want to be able to access these
+   files in a reliable and efficient way.
+3. The Tendermint Core team wants to provide stable interfaces that are as easy
+   as possible to maintain, on which consumers can depend, and to be able to
+   notify those consumers promptly when those interfaces change. To this end, we
+   want to:
+   1. Prevent any breaking changes from being introduced in minor/patch releases
+      of Tendermint. Only major version updates should be able to contain
+      breaking interface changes.
+   2. Ensure that `.proto` files don't contain invalid syntax.
+   3. Ensure that `.proto` files look reasonably similar in format for code
```

Comment:
I think the right balance here is to use `clang-format` to format proto files. Like gofmt, it's easy to set up and the defaults are fine. I don't feel like automation is that important, except maybe at the level of "this file needs formatting".

Reply:
Ah, you said this below.

Reply:
FYI, we're coming out with `buf format` https://github.com/bufbuild/buf/pull/967